### PR TITLE
Add conditional operator before checking length

### PR DESCRIPTION
**Pull request by barbaravaldez**
This PR is for #15690

One of the entries of the toc had a definition of sections, but didn't have any file entries below it.
So sections were not undefined, but when we tried to check the length then it'll throw an error.
Adding an conditional operator so none of the editing operations fail.

### CODE REVIEW

File: extensions/notebook/src/book/bookTocManager.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export interface quickPickResults {
 const allowedFileExtensions: string[] = [FileExtension.Markdown, FileExtension.Notebook];
 
 export function hasSections(node: JupyterBookSection): boolean {
-	return node.sections !== undefined && node.sections.length > 0;
+	return node.sections !== undefined && node.sections?.length > 0;
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTocManager.ts:33]: This shouldn't be needed - you already check that node.sections isn't undefined which the short-circuiting logic should mean that it never evaluates the second condition if that is undefined. Were you seeing differently?

Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTocManager.ts:33]: Yeah, I think the issue is that node.sections is null

**Review by corivera — APPROVED** [review]

**Review by VasuBhog — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/notebook/src/book/bookTocManager.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/notebook/src/book/bookTocManager.ts]: Ah gotcha - is this coming from some external library then? We shouldn't be using null ourselves but this makes sense if it's something we can't control.

Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTocManager.ts]: yes, it's from js-yaml when we parse the toc

**Pull request merged**